Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift

```diff
@@ -59,4 +59,20 @@ public class StarknetProvider: StarknetProviderProtocol {
         
         return result
     }
+    
+    public func getNonce(of contract: Felt, at blockId: StarknetBlockId) async throws -> Felt {
+        let params = GetNonceParams(contractAddress: contract, blockId: blockId)
+        
+        let result = try await execute(method: .getNonce, params: params, receive: Felt.self)
```

Comment:
It existed since before this pr, but I'm unsure if using `execute` is a good method name in provider. Since StarknetAccountProtocol also has an `execute` and it kind of suggests transaction being signed here.

Reply:
execute is internal, but still, good point. I'lll come up with another name for it